core/rfb.js

```diff
@@ -1788,6 +1772,34 @@ RFB.genDES = function (password, challenge) {
 };
 
 RFB.encodingHandlers = {
+    getHandler: function (rfb, encoding) {
+        switch (encoding) {
```

Comment:
Please, let's not do this.  We should have a map of some sort (whether via the Map type or just a Javascript object) and do lookup in there, instead of having a giant switch statement.  If the names of the functions in the encodingHandlers object isn't correct, then we should fix the names.

Reply:
I don't think it's a good idea to restrict the code to follow the naming of protocol constants. But would you prefer something like this in the init routine:

```Javascript
this._encoding_handlers = []
this._encoding_handlers[encodings.encodingRaw] = RFB.encodingHandlers.RAW.bind(this)
...
```

Reply:
yeah.  I'd either prefer that or just having something like
`RFB.encodingHandlers.registerEncoding(encodings.encodingRaw, function(...) { ... })`.  I don't like the giant switch statement, and the ability to register new encodings dynamically is nice.

P.S. won't it be nice when we can drop IE support in favor of edge at some nebulous time in the future and just write `RFB.encodingHandlers = { ..., [encodings.encodingRaw]: function(...) { ... } }`).  All the other browsers support it.

Reply:
One can dream. :)
Unfortunately Edge isn't much better in many regards...

I've pushed a new version now with the suggested model. Please have a look.